docs/peg_oled_display.md

```diff
@@ -0,0 +1,97 @@
+# Peg Oled Display
+To use this you need to make some changes to your kb.py as well as you main.py I will break it into two sections for you. 
+
+### What you can and cant do
+
+#### Can do
+* display images
+* display text
+* set images or text to react to your layer
+
+
+#### Cant do yet / on the way
+* react to battery percentage
+* react to WPM 
+
+## Required Libs
+you need these frozen into your circuitpython or in a lib folder at the root of your drive.
+* [Adafruit_CircuitPython_DisplayIO_SSD1306](https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_SSD1306)
+* [Adafruit_CircuitPython_Display_Text](https://github.com/adafruit/Adafruit_CircuitPython_Display_Text)
+* [download .mpy versions from here](https://github.com/adafruit/Adafruit_CircuitPython_Bundle/releases/download/20220415/adafruit-circuitpython-bundle-7.x-mpy-20220415.zip)
+
+
+## kb.py
+Your chosen board may already have these changes done If not you will need to add them.
+
+you need to add SCL and SDA to your keyboard. The pins the oled are connected to on your controller may not be called "SCL"and "SDA" they could be "GP21" and "GP13" for example. The best way to find out what they are is look at the boards pinout.
+```python
+    SCL=board.SCL
+    SDA=board.SDA
+```
+
+
+## Main.py
+These are the changes that need to be made / added to your main.py
+### Config
+no mater how you are going to use the oled you need this part
+```python
+    from kmk.extensions.peg_oledDisplay import oled
+    keyboard = KMKKeyboard()
+    # ... Other oled code
+    keyboard.extensions.append(oled_ext) 
+
+```
+### Photos
+So the config for photos is quite simple. Getting the photos maybe not so much. I will explain.
+
+oled takes 3-4 arguments 
+1. keyboard
+    * this is where we get the config from
+2.  array of data
+    * this array can have a length of 1 or 4. For images we use a length of 1.
+    * Every item in the array has 2 fields 
+    * 0: this is the reaction type right now it can be "LAYER" or "STATIC"
+    * 1: the items you want to show for the reaction. In this example 4 images to switch on the 4 layers
+3. toDisplay this takes a string "TXT" or "IMG".
+    * this tells the extension to load images or text.
+4. flip Boolean this will simply flip your display.
+
+![code layout](https://i.imgur.com/MrRYzo8.png)
+
+```python
+oled_ext = oled(keyboard, [{0:"LAYER",1:["1.bmp","2.bmp","3.bmp","4.bmp"]}],toDisplay='IMG',flip=True)

```

Comment:
API change: this should not need to take `keyboard` as an init arg, because the Modules and Extensions APIs already pass the keyboard object reference as the first argument to `during_bootup` (https://github.com/KMKfw/kmk_firmware/blob/a210c2bf62c15c70e6af4c42d5b8bf6984bece5b/kmk/kmk_keyboard.py#L422-L424 and https://github.com/KMKfw/kmk_firmware/blob/a210c2bf62c15c70e6af4c42d5b8bf6984bece5b/kmk/kmk_keyboard.py#L429-L431). Initialization routines that require access to the overall keyboard object should be able to defer that access until `on_bootup` is called.